File: code/cpp/robot_control.cpp
/*
 * TEKNOFEST C++ Robot Kontrol Örneği
 * OOP kullanarak robot kontrol sistemi
 */

#include <iostream>
#include <string>

class Robot {
private:
    std::string name;
    int x, y;
    std::string direction;
    
public:
    Robot(const std::string& robotName) : name(robotName), x(0), y(0), direction("Kuzey") {}
    
    void moveForward(int distance = 1) {
        if (direction == "Kuzey") y += distance;
        else if (direction == "Güney") y -= distance;
        else if (direction == "Doğu") x += distance;
        else if (direction == "Batı") x -= distance;
        
        std::cout << name << " " << distance << " birim ileri gitti. Konum: (" 
                  << x << ", " << y << ")" << std::endl;
    }
    
    void turnRight() {
        std::string directions[] = {"Kuzey", "Doğu", "Güney", "Batı"};
        int current = 0;
        for (int i = 0; i < 4; i++) {
            if (directions[i] == direction) {
                current = i;
                break;
            }
        }
        direction = directions[(current + 1) % 4];
        std::cout << name << " sağa döndü. Yön: " << direction << std::endl;
    }
    
    void printPosition() {
        std::cout << "📍 Konum: (" << x << ", " << y << ") Yön: " << direction << std::endl;
    }
};

int main() {
    std::cout << "🤖 TEKNOFEST Robot Kontrol Sistemi\n" << std::endl;
    
    Robot robot("TEKNOFEST-Bot");
    
    robot.moveForward(3);
    robot.turnRight();
    robot.moveForward(2);
    robot.turnRight();
    robot.moveForward(1);
    
    std::cout << "\n";
    robot.printPosition();
    
    return 0;
}

File: code/java/RobotControl.java
/**
 * TEKNOFEST Java Robot Kontrol Örneği
 * OOP ile robot kontrol sistemi
 */

public class RobotControl {
    
    static class Robot {
        private String name;
        private int x, y;
        private String direction;
        
        public Robot(String name) {
            this.name = name;
            this.x = 0;
            this.y = 0;
            this.direction = "Kuzey";
        }
        
        public void moveForward(int distance) {
            switch (direction) {
                case "Kuzey": y += distance; break;
                case "Güney": y -= distance; break;
                case "Doğu": x += distance; break;
                case "Batı": x -= distance; break;
            }
            System.out.println(name + " " + distance + " birim ileri gitti. Konum: (" + x + ", " + y + ")");
        }
        
        public void turnRight() {
            String[] directions = {"Kuzey", "Doğu", "Güney", "Batı"};
            int current = 0;
            for (int i = 0; i < directions.length; i++) {
                if (directions[i].equals(direction)) {
                    current = i;
                    break;
                }
            }
            direction = directions[(current + 1) % 4];
            System.out.println(name + " sağa döndü. Yön: " + direction);
        }
        
        public void printPosition() {
            System.out.println("📍 Konum: (" + x + ", " + y + ") Yön: " + direction);
        }
    }
    
    public static void main(String[] args) {
        System.out.println("🤖 TEKNOFEST Robot Kontrol Sistemi\n");
        
        Robot robot = new Robot("TEKNOFEST-Bot");
        
        robot.moveForward(3);
        robot.turnRight();
        robot.moveForward(2);
        robot.turnRight();
        robot.moveForward(1);
        
        System.out.println();
        robot.printPosition();
    }
}

